### ISSUE TITLE
Middleware does not run on local dev server

### ISSUE THREAD

**Issue by mateusfl** [withastro/astro#7337]
### What version of `astro` are you using?

2.6.1

### Are you using an SSR adapter? If so, which one?

Vercel

### What package manager are you using?

npm

### What operating system are you using?

Linux

### What browser are you using?

Brave

### Describe the Bug

The middleware in my project works fine when deployed on vercel.

However on my machine, when the dev server is running, the middleware does not seem to even be detected, no logs are executed and changes to the `Astro.locals` variable are not acessible in the astro pages.
I've already tried creating a new empty project, switching to pnpm, changing adapters or render modes, and still doesn't seem to work.

**Note:** The middleware on the minimal reproducible example below works fine on stackblitz, as it would probably do on vercel, but downloading the exact project and running on my machine does not work.

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-cwvpnw?file=src%2Fmiddleware.ts

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by mateusfl** [comment]
Node version is 20.0.0
Just installed using nvm and i guess it pulled the latest possible

**Comment by ematipico** [comment]
I can't reproduce it. I just ran the stackblitz example and I could see the log.

**Comment by ematipico** [comment]
@mateusfl I think I am missing something; please answer my questions.
1. The repro doesn't have `experimental: { middleware: true }`, which was required for version `2.5.*`. Is it a mistake?
2. The repro doesn't have any adapter, but on the other hand, you said you use vercel adapter. Is intentional?

**Comment by mateusfl** [comment]
> @mateusfl I think I am missing something; please answer my questions.
> 
> 1. The repro doesn't have `experimental: { middleware: true }`, wh*ich was required for version `2.5.*`. Is it a mistake? Also, you said you used version `2.6.1` but the example uses `2.5.*`.
> 2. The repro doesn't have any adapter, but on the other hand, you said you use vercel adapter. Is it intentional?

Hey @ematipico, sorry about the inconsistencies but i believe all the information in the issue is correct.

The version in the repro was the minimal example generated by astro, the only change is the added middleware file, wich runs fine even without the `experimental: { middleware: true }`, in fact, adding the experimental middleware config breaks the project with an `Invalid experimental key: middleware` error, so i'm not really sure about that behaviour.

The information about the vercel adapter is just to answer the default question of the issue and provide some context about it. The issue happens regardless of using the vercel adapter, or any adapter at all.

**Comment by ematipico** [comment]
Thank you. Without a proper way to reproduce the issue, it would be really difficult for me to help you here.

**Comment by mateusfl** [comment]
> Thank you. With a proper way to reproduce the issue, it would be easier for me to help you here. I created a new project as you suggested, but the middleware works fine for me.

Thank you for taking the time to try, i appreciate it. 

As i understand the issue seems to have a very specific (and sadly unknown) condition to happen. Because it is so difficult to reproduce and has that "happening just with me" feeling to it, i think i will just close the issue as not planned and hope no one stumbles uppon it as i did. Thanks for the support tho.

**Comment by JoeMorgan** [comment]
@mateusfl did you ever figure out your problem? I'm seeing the same thing

**Comment by mateusfl** [comment]
> @mateusfl did you ever figure out your problem? I'm seeing the same thing

I actually did.
After getting to the point of creating a clean virtual machine with my exact system and creating a folder to test the issue and getting it to work, i tested moving the folder where i was creating my astro projects up in the my system's folder tree, like so:
What i had: 
Root -> mateus -> some -> folder -> astro_projects -> project_folder

I moved the folder where i create my projects up 2 levels, like this: 
Root -> mateus -> astro_projects -> project_folder

And it just worked, that was the only change. I don't know the specific reason for it, as the folder the project is now is not even root, is just a place where i organize my stuff, but moving the project to that location solved it for me.

Hope this helps.
